metals/src/main/scala/scala/meta/internal/metals/codeactions/FlatMapToForComprehensionCodeAction.scala

```diff
@@ -0,0 +1,541 @@
+
+      case Term.ApplyUnary(op, arg) if allowedToGetInsideApply =>
+        val (newArg, argReplacementTimes) =
+          replacePlaceHolder(arg, newName, allowedToGetInsideApply)
+        (Term.ApplyUnary(op, newArg), argReplacementTimes)
+
+      case Term.ApplyUsing(fun, args) if allowedToGetInsideApply =>
+        val (newFun, funReplacementTimes) =
+          replacePlaceHolder(fun, newName, false) // TODO false??
+        val (newArgs, argsReplacementTimes) =
+          args.map(replacePlaceHolder(_, newName, false)).unzip // TODO false??
+
+        val replacementTimes =
+          argsReplacementTimes.fold(funReplacementTimes)((result, newElem) =>
+            result + newElem
+          )
+
+        (Term.ApplyUsing(newFun, newArgs), replacementTimes)
+
+      case Term.ApplyInfix(lhs, op, targs, args) if allowedToGetInsideApply =>
+        val (newLHS, lhsReplacementTimes) =
+          replacePlaceHolder(lhs, newName, allowedToGetInsideApply)
+        val (newArgs, argsReplacementTimes) =
+          args
+            .map(replacePlaceHolder(_, newName, allowedToGetInsideApply))
+            .unzip
+
+        val replacementTimes =
+          argsReplacementTimes.fold(lhsReplacementTimes)((result, newElem) =>
+            result + newElem
+          )
+
+        (Term.ApplyInfix(newLHS, op, targs, newArgs), replacementTimes)
+
+      case Term.Select(qual, name) =>
+        val (newQual, qualReplacementTimes) =
+          replacePlaceHolder(qual, newName, allowedToGetInsideApply)
+        (
+          Term.Select(
+            newQual,
+            name
+          ),
+          qualReplacementTimes
+        )
+      case Term.Placeholder() => (newName, 1)
+      case other => (other, 0)
+    }
+
+  private def replacePlaceHolderInTermWithNewName(
+      term: Term,
+      generatedByMetalsValues: Set[String]
+  ): (Option[(String, Term)], Set[String]) = {
+    val (newName, generatedValues) =
+      createNewName(term, generatedByMetalsValues)
+
+    val (newTerm, replacementTimes) =
+      replacePlaceHolder(term, Term.Name(newName), true)
+    (
+      if (replacementTimes == 1) Some((newName, newTerm)) else None,
+      generatedValues
+    )
+  }
+
+  private def processValueNameAndNextQual(
+      tree: Tree,
+      generatedByMetalsValues: Set[String]
+  ): (Option[(String, Term)], Set[String]) = {
+    tree match {
+      case Term.Function(List(param), term) if param.name.value.isEmpty =>
+        val (newName, generatedValues) =
+          createNewName(tree, generatedByMetalsValues)
+        (Some((newName, term)), generatedValues)
+      case Term.Function(List(param), term) =>
+        (Some((param.name.value, term)), generatedByMetalsValues)
+
+      case Term.AnonymousFunction(term) =>
+        replacePlaceHolderInTermWithNewName(term, generatedByMetalsValues)
+      case term: Term.ApplyInfix =>
+        replacePlaceHolderInTermWithNewName(term, generatedByMetalsValues)
+      case term: Term.ApplyUnary =>
```

Comment:
for getting tests to pass

Reply:
I remember that `.map(!negate(_))` was failing for the exact same reason I explained above. That’s why I included these cases in the test.

Reply:
But what about `.map(curriedFunction(1))` ? Doe sit currently work for that case?

Reply:
No, it is not🧐 I would fix that 🙂
Thank you so much for brining it up 👍